src/components/dataTable/js/tableRowDirective.js

```diff
@@ -0,0 +1,326 @@
+ * <table md-table md-row-select md-selected="$ctrl.selected">
+ *   <tbody>
+ *     <tr md-row md-selectable="account.user as user" md-track-by="user.email" ng-repeat="account in $ctrl.accounts">
+ *       <td>{{::user.name}}</td>
+ *     </tr>
+ *   </tbody>
+ * </table>
+ *</hljs>
+ *
+ * <hljs lang="html">
+ *  <!-- this example assigns the result of a function call to an alias named copy -->
+ * <table md-table md-row-select md-selected="$ctrl.selected">
+ *   <tbody>
+ *     <tr md-row md-selectable="$ctrl.copy(item) as copy" md-track-by="copy.id" ng-repeat="item in $ctrl.items">
+ *       <td>{{::copy.name}}</td>
+ *     </tr>
+ *   </tbody>
+ * </table>
+ * </hljs>
+ *
+ * @param {expression=} md-selectable An expression of the form `expression (as alias)?`. The value of the expression will be the item selected.
+ * @param {expression=} md-track-by Optionally use an expression to track items. Otherwise equality will be determend by `===`.
+ * @param {expression=} md-on-select An expression to be executed when an item is selected (will be called for each item).
+ * @param {expression=} md-on-deselect An expression to be executed when an item is deselected (will be called for each item).
+ * @param {boolean=} md-auto-select Allow the user to select a row by clicking anywhere in the row.
+ * @param {boolean=} disabled Will not allow the row to be selected. You may use `ng-disabled` to set this attribute.
+ */
+function MdTableRowDirective() {
+
+  return {
+    restrict: 'A',
+    require: ['mdRow', '?mdSelectable', '?^^mdHead', '^^mdTable'],
+    controller: MdTableRowController,
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs, ctrls) {
+    var ctrl = ctrls[0];
+    var selectCtrl = ctrls[1];
+    var headCtrl = ctrls[2];
+    var tableCtrl = ctrls[3];
+
+    ctrl.initCtrl(tableCtrl, headCtrl, selectCtrl);
+  }
+}
+
+function MdTableRowController($element, $scope) {
+  this.$element = $element;
+  this.$scope = $scope;
+}
+
+MdTableRowController.prototype.initCtrl = function(tableCtrl, headCtrl, selectCtrl) {
+  this.table = tableCtrl;
+  this.head = headCtrl;
+  this.selectCtrl = selectCtrl;
+
+  this.$scope.$watch(this.table.isSelectable.bind(this.table), function(isSelectable) {
+    isSelectable ? this.onEnableSelection() : this.onDisableSelection();
+  }.bind(this));
+};
+
+MdTableRowController.prototype.onEnableSelection = function() {
+  var cellElement = angular.element(this.head ? '<th md-cell>' : '<td md-cell>');
+  // Add a class to identify the cells, which are only temporary present in the DOM, to show the
+  // selection checkbox.
+  cellElement.addClass('md-checkbox-cell');
+
+  this.$element.prepend(cellElement);
+
+  if( this.head) {
+    // We have to call the heads onEnableSelection function, to update the toggle-all checkbox
+    // in the head row.
+    this.head.onEnableSelection();
+  } else if (this.selectCtrl) {
+    //TODO(devversion): revisit the select controller.
```

Comment:
Pretty weird comment from myself. Can't recall why I should revisit that :smile: